src/main/java/com/thebois/models/world/structures/AbstractStructure.java

```diff
@@ -11,27 +15,39 @@
 
     private Position position;
     private StructureType structureType;
+    private IInventory allNeededItems;
+    private IInventory deliveredItems = new Inventory();
 
     /**
      * Creates a structure with a position and structure type.
      *
-     * @param posX          Position in X-axis
-     * @param posY          Position in Y-axis
+     * @param posX          Position in X-axis.
+     * @param posY          Position in Y-axis.
      * @param structureType The type of structure to create.
+     * @param allNeededItems The items needed to finalize construction.
      */
-    AbstractStructure(int posX, int posY, StructureType structureType) {
+    AbstractStructure(
+        int posX,
+        int posY,
+        StructureType structureType,
+        final IInventory allNeededItems) {
         this.position = new Position(posX, posY);
         this.structureType = structureType;
+        this.allNeededItems = allNeededItems;
```

Comment:
See comment about field, then we could skip this line 😄

Reply:
There still needs to be a way to set what items are needed, this should be done by the structure factory

Reply:
👍